lib/vmdb/util.rb

```diff
@@ -1,7 +1,17 @@
 module VMDB
   module Util
-    def self.http_proxy_uri
-      proxy = VMDB::Config.new("vmdb").config[:http_proxy] || {}
+    def self.http_proxy_uri(proxy_config = :default)
+      # TODO: (julian) This looks really messy. Need to make it look nicer.
+      vmdb_proxy = Settings.http_proxy
+
+      return nil unless vmdb_proxy
+
+      if vmdb_proxy[proxy_config].nil?
+        $log.warn("Could not find proxy setting for #{proxy_config}")
+      end
+
+      proxy = vmdb_proxy[proxy_config].to_h || vmdb_proxy.to_h
```

Comment:
This wont work. `vmdb_proxy[proxy_config].to_h` is always a hash. And the `.to_h` is not needed because you can access `Settings` like a hash and with `.methods`. Try it in `rails c`, it's great, like an OpenStruct